web-server/app/codesearch/web/controllers/BaseController.scala

```diff
@@ -0,0 +1,81 @@
+package codesearch.web.controllers
+
+import cats.data.OptionT
+import codesearch.core.db.DefaultDB
+import codesearch.core.index.LanguageIndex
+import codesearch.core.index.LanguageIndex.{CSearchPage, SearchArguments}
+import codesearch.core.model.DefaultTable
+import codesearch.core.util.Helper
+import com.github.marlonlom.utilities.timeago.TimeAgo
+import play.api.mvc.InjectedController
+import cats.instances.future._
+
+import scala.concurrent.{ExecutionContext, Future}
+
+/**
+  * @author sss3 (Vladimir Alekseev)
+  */
+trait BaseController[V <: DefaultTable, I <: LanguageIndex[V]] { self: InjectedController =>
```

Comment:
The trait name misleads cause of it's not base trait for all controllers, but only for search controllers

Reply:
U are right :+1: